components/reference-data/src/main/java/org/datacleaner/beans/ReferenceDataMatcherAnalyzerReducer.java

```diff
@@ -0,0 +1,120 @@
+            createDimensions(columnValueCombinationCrosstabDimensions, partialValueCombinationCrosstab);
+        }
+
+        //create the crosstabs
+        final Crosstab<Number> newResultColumnStatistics = new Crosstab<Number>(Number.class,
+                columnStatisticCrosstabDimensions);
+        final Crosstab<Number> newResultColumnValueCombination = new Crosstab<Number>(Number.class,
+                columnValueCombinationCrosstabDimensions);
+
+        //define the navigators
+        final CrosstabNavigator<Number> columnStatisticsNavigator = new CrosstabNavigator<Number>(
+                newResultColumnStatistics);
+        final CrosstabNavigator<Number> columnValueCombinationNavigator = new CrosstabNavigator<Number>(
+                newResultColumnValueCombination);
+        
+        //add the partial results 
+        for (BooleanAnalyzerResult partialResult : partialResults) {
+            final Crosstab<Number> partialColumnStatisticsCrosstab = partialResult.getColumnStatisticsCrosstab();
+            final Crosstab<Number> partialValueCombinationCrosstab = partialResult.getValueCombinationCrosstab();
+            addData(columnStatisticsNavigator, partialColumnStatisticsCrosstab);
+            addData(columnValueCombinationNavigator, partialValueCombinationCrosstab);
+        }
+
+        return new BooleanAnalyzerResult(newResultColumnStatistics, newResultColumnValueCombination);
+    }
+
+    private void createDimensions(List<CrosstabDimension> crosstabDimensions,
+            final Crosstab<Number> partialColumnCrosstab) {
+        if (partialColumnCrosstab != null) {
+            final List<CrosstabDimension> dimensions = partialColumnCrosstab.getDimensions();
+            for (CrosstabDimension dimension : dimensions) {
+                if (!dimensionExits(crosstabDimensions, dimension)) {
+                    crosstabDimensions.add(dimension);
+                }
+            }
+        }
+    }
+
+    private void addData(final CrosstabNavigator<Number> mainNavigator, final Crosstab<Number> partialCrosstab) {
+        if (partialCrosstab != null) {
+            final CrosstabNavigator<Number> nav = new CrosstabNavigator<Number>(partialCrosstab);
+            final List<CrosstabDimension> dimensions = partialCrosstab.getDimensions();
+            for (CrosstabDimension dimension : dimensions) {
+                final List<String> categories = dimension.getCategories();
+                for (String category : categories) {
+                    final Number categoryValue = nav.where(dimension, category).safeGet(null);
+                    final CrosstabNavigator<Number> whereToPut = mainNavigator.where(dimension, category);
+                    if (categoryValue != null) {
+                        final Number oldValue = whereToPut.safeGet(null);
+                        if (oldValue != null) {
+                            final Number newValue = oldValue.doubleValue() + categoryValue.doubleValue();
```

Comment:
Since we are here calling .doubleValue() we will be creating always a new double value. It's not that big a concern, but maybe we can quickly look to see if there is a utility function somewhere (or roll a basic one ourselves) for the sum function. What I have in mind is that if both numbers are e.g. of type Long, then we return a new Long. Having double as a "fall back" would make sense.